### ISSUE TITLE
Text box constraints

### ISSUE THREAD

**Issue by LordPyrex** [NPBruce/valkyrie#553]
I've noticed that some of the text box constraints are causing fit issues. Is this still ongoing or is this just something that is on my end. Using v1.3

**Comment by NPBruce** [comment]
Are you talking about in the editor or while playing?  Which boxes?  in 1.4 I am changing the editor text boxes to grow/shrink to fit the text size.

**Comment by LordPyrex** [comment]
In play. I'm sure with the next roll out it won't be a problem.

**Comment by NPBruce** [comment]
I am not aware of any issues in play, if you can reproduce one please tell me where, and and what resolution you are running.

**Comment by LordPyrex** [comment]
![moveboxoutofplace](https://user-images.githubusercontent.com/27811809/27616332-0be2f550-5b7d-11e7-92f9-a6c019c42e2a.png)
I'm not sure if it's just quantity based, but I was able to reproduce on 1920x1200 windowed and 3840x2160 Fullscreen

**Comment by LordPyrex** [comment]
![constraint](https://user-images.githubusercontent.com/27811809/27616369-4fd48eea-5b7d-11e7-9e18-c38a7fe3d387.png)
I'm not sure if it's just quantity based, but I was able to reproduce on 1920x1200 windowed and 3840x2160 Fullscreen

**Comment by NPBruce** [comment]
Can you provide the scenario? I don't have a monitor that big so I will do what I can do find out what is going on.

**Comment by LordPyrex** [comment]
[InPlacesUnseen.valkyrie.zip](https://github.com/NPBruce/valkyrie/files/1107301/InPlacesUnseen.valkyrie.zip)
I'm still finishing my text, but it will be ready for the store soon.

**Comment by NPBruce** [comment]
I just want the event text to reproduce the problem... though it may be tricky if it doesn't occur at lower resolutions.

**Comment by LordPyrex** [comment]
Just the event.ini?

**Comment by LordPyrex** [comment]
Or the actual text from the event

**Comment by LordPyrex** [comment]
I've been able to eliminate it by adding a leading blank line, but noticed it happen on other events. I'll check the next alpha to see how that affects things before doing some over hauling

**Comment by NPBruce** [comment]
The development branch has no changes in this area.

**Comment by EtiCarasoft** [comment]
I've found this problem using 1.3 version too but in spanish version for the lower menu:
1440x900 Windowed 
![captura de pantalla 2017-06-28 a les 19 35 14](https://user-images.githubusercontent.com/26690018/27651568-16434c04-5c39-11e7-8d9a-60e7b49addd2.png)

**Comment by redwolf2** [comment]
@EtiCarasoft 

You can change the ...
https://github.com/NPBruce/valkyrie/blob/development/unity/Assets/Resources/Text/Localization.txt

... and have to change:
ITEMS_SMALL and SET

**Comment by LordPyrex** [comment]
@NPBruce I think that this has been resolved with any current iterations. I feel at this point this thread could be closed unless there are other issues floating around out there.

**Comment by NPBruce** [comment]
@LordPyrex I haven't changed anything here since 1.3, so I am not sure how/why this is fixed.  Are you no longer able to produce the issue shown in the screen shot above?

**Comment by LordPyrex** [comment]
Well, as quickly as I said that I noticed another box with this issue.... never mind for now...

**Comment by Indoy** [comment]
Hi!
I also have some trouble to fit correctly text window with my translations. Most of the time it's out the frame. I did not catch if there's any issue for that?
Thank you
![image](https://user-images.githubusercontent.com/33460987/33489375-fc2e986e-d6b3-11e7-856a-d1244f68c76b.png)

**Comment by NPBruce** [comment]
In this case that is the maximum size available.

The scenario could be reworked to split this into two.

**Comment by redwolf2** [comment]
Can't we just show scrollbars on demand?

**Comment by NPBruce** [comment]
No because of the black text unity bug. In general though, I'm not sure that such a wall of text in a single event is a great idea.

If someone wants this add as a feature request (this is not related to this issue).

**Comment by Indoy** [comment]
OK! nevermind, in the future i'll ensure to the translated text fill the box... even if i have to cut a little the original idea!
Thx

**Comment by bluefishj** [comment]
I've had the same issues with my scenario, The Experiment. I was had not looked it up and was hoping it was a known issues that would be fixed in 2.0, but I'm still able to reproduce it. I've tested it on different resolutions on Mac, PC, and Android, all able to reproduce. Start scenario, two token in middle of room, dagger ->  stab -> triggers EventInvestigatorAwakes, takes less than a minute to reproduce. If I open the event, add and remove a space in the text box, then save and test, it fixes it, but doesn't fix it for the exported package. It will stay fixed if I go play in the app right after using the editor, but if I quit the app an then reopen, the text has problems again. I think there were at least a few instances of this when I was testing, but didn't write them down at the time. I know it happens again when you threaten the first NPC.

![screen shot 2018-05-06 at 11 59 08 pm](https://user-images.githubusercontent.com/38113646/39685682-e9a37844-5189-11e8-9be8-41c31f039c23.png)

**Comment by Bakajao** [comment]
Hi all...i had the same problem about the boxes text fit...i investigated why happens and i discover that it fix changing the languaje in quest.ini file....
I hope that solution solve the problems

**Comment by BenRQ** [comment]
Hi @Bakajao, I have the same issue and I tried to change the default language but it did not fix the issue.
I set DefaultLanguage to French, and my text is in French.
Valkyrie version 2.0.1

Can you give more details on your fix ?

![momtextsize](https://user-images.githubusercontent.com/40794717/42537834-52994074-8496-11e8-87c2-a46b4d5482de.png)

**Comment by BenRQ** [comment]
@NPBruce @redwolf2 I have a fix to propose on this one.

It seems that Unity may not provide a correct height when using GetPreferredHeight in some cases.
I tried to attach the object and use Canvas.ForceUpdateCanvases(); to see if it was related but it's not.

It does work fine when using the UIElement that you are going to display though.
So I changed the functions so they are not static anymore, but using the Object itself.
And it works just fine now :)

![image](https://user-images.githubusercontent.com/40794717/42650784-a1d3c330-860d-11e8-9ec3-4b279cb4f82b.png)

Tell me if you are happy with this fix, I will pull this to development.
